tests/tests_pytorch/trainer/connectors/test_data_connector.py

```diff
@@ -271,12 +274,24 @@ class CustomDummyObj:
     assert result.dummy_kwarg is None
 
     class CustomSampler(Sampler):
-        pass
+        def __init__(self, data_source: Sized) -> None:
+            super().__init__(data_source)
+            self.data_source = data_source
+
+        def __len__(self):
+            return len(self.data_source)
+
+        def __iter__(self):
+            return iter(range(len(self)))
 
     # Should raise an error if existing sampler is being replaced
     dataloader = CustomDataLoader(dataset, sampler=CustomSampler(dataset))
-    with pytest.raises(MisconfigurationException, match="will be replaced by `DistributedSampler`"):
-        trainer._data_connector._prepare_dataloader(dataloader, shuffle=True)
+    result = trainer._data_connector._prepare_dataloader(dataloader)
+    result_dataset = list(iter(result))
```

Comment:
I don't think this is requried

```suggestion
    result_dataset = list(result)
```

Reply:
And here as well :)